phpseclib/Math/BigInteger.php

```diff
@@ -752,18 +750,17 @@ function __toString()
     /**
      * __clone() magic method
      *
-     * Although you can call BigInteger::__toString() directly in PHP5, you cannot call BigInteger::__clone() directly
-     * in PHP5.  You can in PHP4 since it's not a magic method, but in PHP5, you have to call it by using the PHP5
-     * only syntax of $y = clone $x.  As such, if you're trying to write an application that works on both PHP4 and
-     * PHP5, call BigInteger::copy(), instead.
-     *
      * @access public
-     * @see self::copy()
      * @return \phpseclib\Math\BigInteger
      */
     function __clone()
     {
-        return $this->copy();
+        $temp = new static();
+        $temp->value = $this->value;
+        $temp->is_negative = $this->is_negative;
+        $temp->precision = $this->precision;
+        $temp->bitmask = $this->bitmask;
+        return $temp;
     }
```

Comment:
No - it's needed. The following demonstrates why:

```php
<?php
include('Math/BigInteger.php');

$a = new Math_BigInteger(5);
$b = $a;
$b->setPrecision(4);

echo $a->precision;
```
That'll output 4 when it ought to output -1.

clone may do a shallow copy but it none-the-less does a copy of the primitives. PHP4 cloned every time you assigned an object to a new variable - PHP5 doesn't. So to make code that worked on both the `copy()` method was added since `__clone()` couldn't be called directly on PHP5. But that all changed when phpseclib introduced namespacing. That upped the version requirement to 5.3 and eliminated the need for a `copy()` method to emulate `__clone()`.

http://www.php.net/manual/en/migration5.oop.php elaborates:

> In previous versions of PHP, objects were handled like primitive types (for instance integers and strings). The drawback of this method was that semantically the whole object was copied when a variable was assigned, or passed as a parameter to a method. In the new approach, objects are referenced by handle, and not by value (one can think of a handle as an object's identifier).

Altho that said, that example does make me think a `getPrecision` method is needed lol